File: ChatMidWare/src/main/java/com/niit/restcontroller/BlogController.java
package com.niit.restcontroller;

import java.util.List;

import javax.servlet.http.HttpSession;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.niit.dao.BlogDAO;
import com.niit.model.Blog;
import com.niit.model.BlogComment;

@RestController
public class BlogController {
	@Autowired
	BlogDAO blogDAO;
		@GetMapping(value="/demo")
		public ResponseEntity<String> demo()
		{
			return new  ResponseEntity<String>("Demo data",HttpStatus.OK);
		}
		@GetMapping(value="/listBlogs")
		public ResponseEntity<List<Blog>>  getListBlogs()
		{
			List<Blog> listBlogs=blogDAO.listAllBlog();
			if(listBlogs.size()>0)
			{
			return new ResponseEntity<List<Blog>>(listBlogs,HttpStatus.OK);
			}
			else
			{
				return new ResponseEntity<List<Blog>>(listBlogs,HttpStatus.NOT_FOUND);
			}
		}
		@PostMapping(value="/addBlog")
		public ResponseEntity<String> addBlog(@RequestBody Blog blog, HttpSession session)
		{
		   blog.setCreatedate(new java.util.Date());		 
			blog.setLikes(0);
			blog.setUsername((String)session.getAttribute("userName"));
			blog.setStatus("NA");
		 if (blogDAO.addBlog(blog))
		 {
			 return new  ResponseEntity<String>("Success",HttpStatus.OK);
			}
		 else
		 {
			 return new  ResponseEntity<String>("Failure",HttpStatus.NOT_FOUND);
		 }
		}
		
		@RequestMapping(value = "/getById/{blogId}", method = RequestMethod.GET)
	    public ResponseEntity<Blog> get(@PathVariable("blogId") int blogId){
	        
	        Blog blog = blogDAO.getBlog(blogId);

	        if (blog == null){
	           
	            return new ResponseEntity<Blog>(HttpStatus.NOT_FOUND);
	        }

	        return new ResponseEntity<Blog>(blog, HttpStatus.OK);
	    }
		
		@RequestMapping(value = "/UpdateBlog/{blogId}" , method = RequestMethod.PUT)
		    public ResponseEntity<Blog> update(@PathVariable("blogId") int blogId, @RequestBody Blog blog){
		       System.out.println(blog.getBlogId());
			   Blog blogs = blogDAO.getBlog(blogId);

		        if (blogs == null){
		            System.out.println("Blog update not possible");
		            return new ResponseEntity<Blog>(HttpStatus.NOT_FOUND);
		        }

		                     blogs.setBlogContent(blog.getBlogContent()); 
		                    
		                       blogDAO.updateBlog(blogs);
		        
		        return new ResponseEntity<Blog>(blogs, HttpStatus.OK);
		    }
		 @RequestMapping(value ="/deleteBlog/{blogId}",method=RequestMethod.DELETE)
		 public ResponseEntity<String> deleteBlog(@PathVariable("blogId") int blogId) 
		 {
		       
			   Blog blogs = blogDAO.getBlog(blogId);

		        if (blogs == null){
		            System.out.println("Sadly not deleted");
		            return new ResponseEntity<String>("Failure",HttpStatus.NOT_FOUND);
		        }

		                  
		                    
		                       blogDAO.deleteBlog(blogs);
		        
		        return new ResponseEntity<String>("Success", HttpStatus.OK);
		    }
		 
		 @RequestMapping(value = "/approve/{blogId}", method = RequestMethod.PUT)
		    public ResponseEntity<Blog> approve(@PathVariable("blogId") int blogId){
		       
			   Blog blogs = blogDAO.getBlog(blogId);

		        if (blogs == null){
		            
		            return new ResponseEntity<Blog>(HttpStatus.NOT_FOUND);
		        }

		        blogs.setStatus("A");
				blogDAO.approveBlog(blogs);
		        return new ResponseEntity<Blog>(blogs, HttpStatus.OK);
		    }
		 @RequestMapping(value = "/rejectBlog/{blogId}", method = RequestMethod.PUT)
		    public ResponseEntity<Blog> reject(@PathVariable("blogId") int blogId){
		       
			   Blog blogs = blogDAO.getBlog(blogId);

		        if (blogs == null){
		            
		            return new ResponseEntity<Blog>(HttpStatus.NOT_FOUND);
		        }

		        blogs.setStatus("NA");
				blogDAO.rejectBlog(blogs);
		        return new ResponseEntity<Blog>(blogs, HttpStatus.OK);
		    }
		 @GetMapping(value="/incrementLikes/{blogId}")
	     public ResponseEntity<String>IncrementsLikes(@PathVariable("blogId")int blogId)
	     {
			 Blog blog=blogDAO.getBlog(blogId);
			 if(blogDAO.incrementLikes(blog))
			 {
				 return new ResponseEntity<String>("Success",HttpStatus.OK);
			 }
			 else
			 {
				 return new ResponseEntity<String>("Failure",HttpStatus.NOT_FOUND);
			 }
	     }
		 @PostMapping(value="/addBlogComment/{blogId}"  )
		  	public ResponseEntity<String>addblogcomment(@RequestBody BlogComment blogComment,@PathVariable("blogId")int blogId, HttpSession session)
		  	{
		  		System.out.println("in add comment control");
		  		blogComment.setBlogId(blogId);
		  		blogComment.setCommentDate(new java.util.Date());       
		  	    blogComment.setUsername(((String)session.getAttribute("userName")));
		  	
		   		
		  				
		  				
		  		if(blogDAO.addBlogComment(blogComment))
		  		{
		  			return new ResponseEntity<String>("Success",HttpStatus.OK);
		  		}
		  		else
		  		{
		  			return new ResponseEntity<String>("Failure",HttpStatus.NOT_FOUND);
		  		}
		  		
		  		
		  	}
		 @RequestMapping(value ="/deleteBlogComment/{commentId}",method=RequestMethod.DELETE)
	 	 public ResponseEntity<String> deleteBlogcomment(@PathVariable("commentId") int commentId) 
	 	 {
	 	       
	 		   BlogComment comments = blogDAO.getBlogComment(commentId);

	 	        if (comments == null){
	 	            
	 	            return new ResponseEntity<String>("Failure",HttpStatus.NOT_FOUND);
	 	        }

	 
	 	                    
	 	                       blogDAO.deleteBlogComment(comments);
	 	        
	 	        return new ResponseEntity<String>("Success", HttpStatus.OK);
	 	    }
	      
		 
		 @RequestMapping(value = "/getByCommentId/{commentId}", method = RequestMethod.GET)
	      public ResponseEntity<BlogComment> getComment(@PathVariable("commentId") int commentId){
	          
	          BlogComment comments = blogDAO.getBlogComment(commentId);

	          if (comments == null){
	             
	              return new ResponseEntity<BlogComment>(HttpStatus.NOT_FOUND);
	          }

	          return new ResponseEntity<BlogComment>(comments, HttpStatus.OK);
	      }
		 @GetMapping(value="/listBlogComments/{blogid}")
		  	public ResponseEntity<List<BlogComment>> getListBlogComment(@PathVariable("blogid")int blogid)
		  	{
		  		
		  		List<BlogComment> listBlogComments=blogDAO.listBlogComment(blogid);
		  		return new ResponseEntity<List<BlogComment>>(listBlogComments,HttpStatus.OK);
		  		
		  	}
		      
		 
}

File: ChatFrontend/WebContent/c_blog/BlogController.js
myApp.controller("BlogController", function($scope, $http, $location,$route) {
	console.log("In BlogController");

	$scope.blog = {
		"blogName" : '',
		"blogContent" : '',
		"createDate" : '',
		"likes" : 0,
		"userName" : '',
		"status" : ''
	}
	

	$scope.blogData;
	
	$scope.blogcomment={
			
			"blogId":'',
			"commentDate":'',
		    "username":'',			
			"commentText":''
					
	}
	$scope.blogcommentData;
	

	$scope.insertBlog = function() {
		alert("In insert blog");
		console.log("In insertBlog method");
		$http.post('http://localhost:8081/ChatMidWare/addBlog',
				$scope.blog).then(fetchAllBlogs(), function(response) {
					$route.reload();
					console.log('Status Text ' + response.statusText);
			
		});
	};

	$scope.deleteBlog = function(blogId){
		//alert("in delete blog");
		$http.delete('http://localhost:8081/ChatMidWare/deleteBlog/'+blogId)
		.then(fetchAllBlogs(), function(response){
			console.log('Blog deleted '+ blogId);
		    console.log('Response Status ' + response.statusText);
		    $route.reload();
			$location.path('/listBlogs');
			
		});
	};
	
	$scope.incrementLike=function(blogId)
	{
            	//console.log('Enter into the incrementLike blog method');
		        $http.get('http://localhost:8081/ChatMidWare/incrementLikes/'+blogId)
		.then( fetchAllBlogs(),function(response)
		{
			  console.log('Incerement');
			  $route.reload();
			  $location.path('/listBlogs');			         
			        
				});
	};
	
	
	$scope.editBlog = function(blogId){

		$http.get('http://localhost:8081/ChatMidWare/getById/'+blogId)
		.then(function(response){
			console.log('In edit blog');
			console.log(blogId);
			$scope.blog = response.data;			
			
		});
	};
	
	$scope.updateBlog=function(blogId)
	{
		console.log('Enter into the update blog method');
		console.log(blogId);
		$http.put('http://localhost:8081/ChatMidWare/UpdateBlog/'+blogId,$scope.blog)
		.then(fetchAllBlogs(),function(response)
				{
			
			        console.log('Status Text:'+response.statusText);
			        $route.reload()			   
			        $location.path("/listBlogs");
			      
				});
	};
	$scope.approveBlog=function(blogId)
	{
		console.log('Enter into the approve blog method');
		console.log(blogId);
		$http.put('http://localhost:8081/ChatMidWare/approve/'+blogId)
		.then(function(response)
				{
			
			        console.log('Status Text:'+response.statusText);
			        $route.reload();
					  $location.path('/listBlogs');
			   
			      
				});
	};
	$scope.rejectBlog=function(blogId)
	{
		console.log('Enter into the reject blog method');
		console.log(blogId);
		$http.put('http://localhost:8081/ChatMidWare/rejectBlog/'+blogId)
		.then(function(response)
				{
			
			        console.log('Status Text:'+response.statusText);
			        $route.reload();
					  $location.path('/listBlogs');
			   
			      
				});
	};
	
	
	function fetchAllBlogs() {
		console.log('In fetchAllBlogs method');
		$http.get('http://localhost:8081/ChatMidWare/listBlogs').then(
				function(response) {
					//console.log('Status Text ' + response.statusText);
					$scope.blogData = response.data;
				});
	};
	$scope.listblogComments = function(blogId){

		$http.get('http://localhost:8081/ChatMidWare/listBlogComments/'+blogId)
		.then(function(response){
			console.log('In edit blog');
			console.log(blogId);
			$scope.blogcommentData = response.data;			
			//$location.path('/changeBlog');
		});
	};
	$scope.deleteBlogComment = function(commentId){
		//alert("in delete blog");
		$http.delete('http://localhost:8081/ChatMidWare/deleteBlogComment/'+commentId)
		.then(fetchAllBlogs(), function(response){
			
		    console.log('Response Status ' + response.statusText);
		    $route.reload();
			
			
		});
	};
	
	
	
	
	$scope.insertblogComment = function(blogId) {
		alert("In insert blog comment ");
		
		$http.post('http://localhost:8081/ChatMidWare/addBlogComment/'+blogId,
				$scope.blogcomment).then( function(response) {
					console.log('Status Text ' + response.statusText);
					$route.reload();					
					$location.path('/listBlogs');
			
		});
	};
	
	/*function fetchAllBlogComments(blogId) {
		console.log('In coomment method');
		$http.get('http://localhost:8081/ChatMidWare/listBlogComments/{blogid}').then(
				function(response) {
					//console.log('Status Text ' + response.statusText);
					$scope.blogcommentData = response.data;
				});
	};*/
	
	
	fetchAllBlogs();
});